xbmc/addons/addoninfo/AddonInfoBuilder.cpp

```diff
@@ -409,15 +409,24 @@ bool CAddonInfoBuilder::ParseXMLTypes(CAddonType& addonType, AddonInfoPtr info,
     if (library != nullptr)
     {
       addonType.m_libname = library;
-      // linux is different an has the version number after the suffix
-      static const std::regex libRegex("^.*" +
-                                       CCompileInfo::CCompileInfo::GetSharedLibrarySuffix() +
-                                       "\\.?[0-9]*\\.?[0-9]*\\.?[0-9]*$");
-      if (std::regex_match(library, libRegex))
+
+      try
+      {
+        // linux is different and has the version number after the suffix
+        static const std::regex libRegex("^.*" +
+                                        CCompileInfo::CCompileInfo::GetSharedLibrarySuffix() +
+                                        "\\.?[0-9]*\\.?[0-9]*\\.?[0-9]*$");
+        if (std::regex_match(library, libRegex))
+        {
+          info->SetBinary(true);
+          CLog::Log(LOGDEBUG, "CAddonInfoBuilder::{}: Binary addon found: {}", __FUNCTION__,
```

Comment:
I prever to use C++11 `__func__` instead of `__FUNCTION__`, as Windows include also the class name by it and then bring  **"CAddonInfoBuilder::CAddonInfoBuilder::ParseXMLTypes: ..."**.

See e.g. https://stackoverflow.com/a/45464866

Reply:
Fixed